src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -74,7 +74,10 @@ public override void Initialize(AnalysisContext analysisContext)
 
                     blockStartContext.RegisterOperationAction(operationContext =>
                     {
-                        isInstanceReferenced = true;
+                        if (((IInstanceReferenceOperation)operationContext.Operation).ReferenceKind == InstanceReferenceKind.ContainingTypeInstance)
```

Comment:
don't need to check whether instance is pointing to the type the method belong to?

Reply:
`ContainingTypeInstance` is used for `this` or `base`, both of which are candidates for us to consider method as requiring instance access.